File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/CommunicationRequest.ts
import { Annotation } from "./Annotation";
import { Attachment } from "./Attachment";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type CommunicationRequestPayload = BackboneElement & {
    content?: string | Reference | Attachment;
};

export type CommunicationRequest = DomainResource & {
    category?: CodeableConcept[];
    payload?: CommunicationRequestPayload[];
    encounter?: Reference;
    medium?: CodeableConcept[];
    recipient?: Reference[];
    reasonCode?: CodeableConcept[];
    statusReason?: CodeableConcept;
    authoredOn?: string;
    note?: Annotation[];
    requester?: Reference;
    priority?: string;
    status: string;
    groupIdentifier?: Identifier;
    sender?: Reference;
    identifier?: Identifier[];
    doNotPerform?: boolean;
    replaces?: Reference[];
    basedOn?: Reference[];
    subject?: Reference;
    occurrence?: Period | string;
    about?: Reference[];
    reasonReference?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/CommunicationRequest.py
from typing import Optional, List
from pydantic import *
from ..base import *

class CommunicationRequest_Payload(BackboneElement):
    content_string: Optional[str] = None
    content_reference: Optional[Reference] = None
    content_attachment: Optional[Attachment] = None

class CommunicationRequest(DomainResource):
    category: Optional[List[CodeableConcept]] = None
    payload: Optional[List[CommunicationRequest_Payload]] = None
    encounter: Optional[Reference] = None
    medium: Optional[List[CodeableConcept]] = None
    recipient: Optional[List[Reference]] = None
    reason_code: Optional[List[CodeableConcept]] = None
    status_reason: Optional[CodeableConcept] = None
    authored_on: Optional[str] = None
    note: Optional[List[Annotation]] = None
    requester: Optional[Reference] = None
    priority: Optional[str] = None
    occurrence_period: Optional[Period] = None
    status: str
    group_identifier: Optional[Identifier] = None
    sender: Optional[Reference] = None
    identifier: Optional[List[Identifier]] = None
    do_not_perform: Optional[bool] = None
    replaces: Optional[List[Reference]] = None
    based_on: Optional[List[Reference]] = None
    occurrence_date_time: Optional[str] = None
    subject: Optional[Reference] = None
    about: Optional[List[Reference]] = None
    reason_reference: Optional[List[Reference]] = None